### DISCUSSION TITLE
Having issues with nx/multi application component testing setup

### DISCUSSION THREAD

**Discussion by imdurdydan** [cypress-io/cypress#23594] [Component Testing] [answered]
I am running into trouble when trying to configure cypress component testing for an angular project using nx that is using v2 of the angular json schema. When running cypress open in one of the angular applications and setting up component testing I get this error when it starts trying to initialize the config:

Could not find a project with projectType "application" in "angular.json. 

It seems like cypress component testing does not play nicely with the version 2 of angular.json schema or im doing something incredibly wrong :) (probably the second).

Any insight would be appreciated!

My angular.json:
{ "$schema": "./node_modules/nx/schemas/workspace-schema.json", "version": 2, "projects": { "accelerator-engage": "apps/accelerator-engage", "accelerator-engage-e2e": "apps/accelerator-engage-e2e", "accelerator-message": "apps/accelerator-message", "accelerator-message-e2e": "apps/accelerator-message-e2e", "accelerator-segment": "apps/accelerator-segment", "accelerator-segment-e2e": "apps/accelerator-segment-e2e", "accelerator-shell": "apps/accelerator-shell", "accelerator-shell-e2e": "apps/accelerator-shell-e2e", "cross-product-services": "libs/cross-product-services", "cross-product-types": "libs/cross-product-types", "dashboard": "apps/dashboard", "dashboard-e2e": "apps/dashboard-e2e", "design-system": "libs/design-system", "design-system-e2e": "apps/design-system-e2e" } }

Each application has its own project.json which from what I gather is best practice for nx and multi application angular projects

**Comment by ZachJW34** [comment, marked as answer]
We've just released [v10.7.0](https://docs.cypress.io/guides/references/changelog#10-7-0) which includes a new option in your `cypress.config.ts` that allows you to customize the options passed into the angular dev-server, see https://github.com/cypress-io/cypress/issues/23161.

We've also been working closely with the Nx team to bring Angular CT testing to nx workspaces. You should be able to utilize the new `projectConfig`, but Nx should have a first class API soon.

**Reply by imdurdydan** [reply]
Awesome thank you!
